nbconverted/3_park_analysis.py

```diff
@@ -391,3 +413,277 @@ def get_counts(status):
 # Looking at the box plots, we can see that in general, the samples with "both" a point mutation and a CNV in the gene of interest tend to score higher using our classifiers than samples with "one" of a point mutation or a CNV. This is despite the fact that our classifiers were trained using all of these samples ("one" or "both") as positive labels.
 # 
 # The next step is to break this down by gene - are there genes/cancer types where the functional effect of the "two hits" is clearer, or less clear? Are there genes where we see "two hits" at the genetic level but no functional effect/classifier difference, or are the sources of information basically redundant?
+
+# ### Statistical testing for individual gene/cancer type combinations
+
+# In[26]:
+
+
+info_compare_df = ut.test_all(park_info_df)
+print(info_compare_df.shape)
+print('reject null for:', info_compare_df.reject_null.sum(), '/', info_compare_df.shape[0])
+info_compare_df.sort_values(by='corr_pval', ascending=True).head()
+
+
+# In[27]:
+
+
+# plot top three most significant
+# and top three least significant
+order = ['none', 'one', 'both']
+def get_counts(status):
+    un = np.unique(status, return_counts=True)
+    return {s: c for s, c in zip(*un)}
+
+sns.set({'figure.figsize': (24, 16)})
+fig, axarr = plt.subplots(2, 3)
+
+valid_df = info_compare_df[info_compare_df.p_value != 1.0]
+plot_ids = (
+    valid_df.sort_values(by='corr_pval', ascending=True).identifier.tolist()[:3] + 
+    valid_df.sort_values(by='corr_pval', ascending=False).identifier.tolist()[:3]
+)
+
+for ix, identifier in enumerate(plot_ids):
+    ax = axarr[ix // 3, ix % 3]
+    plot_df = park_info_df[park_info_df.identifier == identifier]
+    sns.boxplot(data=plot_df, x='status', y='positive_prob', order=order, ax=ax)
+    count_map = get_counts(plot_df.status.values)
+    ax.set_xticks(np.arange(3),
+               ['{} (n={})'.format(l, count_map[l]) for l in order])
+    ax.set_title(identifier)
+
+
+# ### Compare classifier-based statistical testing vs. Park et al statistical testing
+
+# In[28]:
+
+
+pair_df = (info_compare_df
+    .merge(park_df, left_on='identifier', right_index=True)
+    .rename(columns={'corr_pval': 'classifier_pval',
+                     'FDR': 'park_pval'})
+)
+print(pair_df.shape)
+print(pair_df['class'].unique())
+print(pair_df.classifier_pval.isna().sum())
+pair_df.head()
+
+
+# In[29]:
```

Comment:
What's with the top line of dots on this plot? Why are so many p-values in the Kim paper at 0.9?

Reply:
Is it to do with their FDR threshold? But still, why are there so dots many right at that x=0.85-0.9 line instead of a gradual increase or an even distribution?

Reply:
I looked at the raw data from the Park et al. paper, and the bunched up p-values seem to be a bunch of repeated values around 0.877. I think this tends to happen with the Benjamini-Hochberg correction (which they used here) - see, e.g. [this description of why repeated values happen](https://stats.stackexchange.com/a/263981).

Looking at the uncorrected p-values, they seem to be more uniform between 0 and 1.